### PR TITLE
feat: index carbon components

### LINKED ISSUE #515: Index Carbon Components

### PR DESCRIPTION
**Pull request by andreancardona**
Closes https://github.com/carbon-design-system/carbon-platform/issues/515

### CODE REVIEW

File: packages/carbon-components/carbon.yml
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]:
```suggestion
  externalDocsUrl: https://the-carbon-components.netlify.app
```

File: packages/carbon-components/carbon.yml
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]: i
```suggestion
          url: https://the-carbon-components.netlify.app/?nav=accordion
```

File: packages/carbon-components/carbon.yml
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]:
```suggestion
```

File: packages/carbon-components/carbon.yml
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]:
```suggestion
```

File: packages/carbon-components/carbon.yml
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]:
```suggestion
      framework: vanilla
```

**Review by francinelucca — CHANGES_REQUESTED** [review]

**Review by francinelucca — COMMENTED** [review]
LGTM, can see 46 components as expected: 
image

And library info:
image

This should be good to go once we resolve remaining TODOs (get final thumbnails and tag approval) @andreancardona  👍 

Let's also confirm treeview status @andreancardona since it's unstable on styles and it's not present in storybook

File: packages/carbon-components/carbon.yml
Comment (aubrey-oneal) [inline packages/carbon-components/carbon.yml]: Approved 👍

File: packages/carbon-components/carbon.yml
Comment (aubrey-oneal) [inline packages/carbon-components/carbon.yml]: Remove this component.

File: packages/carbon-components/carbon.yml
Comment (aubrey-oneal) [inline packages/carbon-components/carbon.yml]: Approved 👍

**Review by aubrey-oneal — CHANGES_REQUESTED** [review]
Just remove the menu component.

**Review by francinelucca — APPROVED** [review]
Lib and comps being pulled correctly:
image
LGTM

**Pull request closed without merging**
